proposals/spi-proposal.md

```diff
@@ -112,168 +105,150 @@ MSBFIRST = 1
-                            0x00: default behavior (as specified by csPinOptions)
-                            0x01: CS_DISABLE
-                            0x02: CS_START_ONLY
-                            0x03: CS_END_ONLY
-5.  numWords              (number of words to transfer)
-6:  data 0                (bits 0-6)
-7:  data 0                (bits 7-14 if word size if word size > 7 && < 15)
-8:  data 0                (if word size > 14)
+2:  SPI_TRANSFER          (0x02)
+3:  deviceId | channel    (bits 3-6: deviceId, bits 0-2: channel)
+4:  requestId             (0-127) // increment for each call
+5:  csPinChange           (0 = don't deselect csPin
+                           1 = deselect csPin (default))
+6.  numWords              (0-127: number of words to transfer)
+7:  data 0                (bits 0-6)
+8:  data 0                (bits 7-14 if word size if word size > 7 && < 15)
+9:  data 0                (if word size > 14)
 ...                       up to numWords * (wordSize / 7)
 N:  END_SYSEX
 ```
 
-#### csPinControl
-
-If a chip select (CS) pin is specified in the `SPI_BEGIN_TRANSACTION` message, the
-default behavior is to set the CS active state at the beginnnig of the transfer and
-toggle the CS pin at the end of the transfer. Use the `csPinControl` to override the
-default behavior. The primay use case for `csPinControl` is breaking up a large amount
-of data over multipe transfers (for example due to buffer size limitiations). Use the
-options `CS_DISABLE`, `CS_START_ONLY` and `CS_END_ONLY` as describe below.
-
-`0x00: DEFAULT` Set `CS_ACTIVE_STATE` at beginning of transfer and toggle CS pin at end
-unless `CS_TOGGLE` is also set (in that case toggle between every word).
-
-`0x01: CS_DISABLE` Disable CS pin control (even if a CS pin was specified in the
-`SPI_BEGIN_TRANSACTION` command). Useful to avoid toggling at the end of middle transfers
-in when a large amount of data requires multiple transfers.
-
-`0x02: CS_START_ONLY` Set active state only at beginning of transfer. Useful if breaking
-up a transfer across multiple packets.
-
-`0x03: CS_END_ONLY` Toggle active state only at end of transfer. Useful if breaking up a
-transfer across multiple packets.
+### SPI_WRITE
 
+Only write data, ignoring any data returned by the slave device.
 
-### SPI_WRITE
+Provided as a convenience. The same can be accomplished using `SPI_TRANSFRER`
+and ignoring the `SPI_REPLY` command.
 
-Only write data, ignoring any data returned by the slave device. No `SPI_REPLY` message
-is sent.
+If `CS_PIN_CONTROL` is enabled, then the CS pin active state will be set when
+the `SPI_WRITE` command is received. It will only be deselected (toggled) at the
+end of the write if `csPinChange` is set to 1.
 
-Provided as a convenience. The same can be accomplished using `SPI_TRANSFRER` and ignoring
-the `SPI_REPLY` message.
+A `SPI_WRITE` command should return a `SPI_REPLY` with a value of `1` if the
```

Comment:
This makes me wonder if we need a way to signal an error for `SPI_READ` and `SPI_TRANSFER` commands as well. One approach is to return a single value `0` if there is an error. However that does not work if the read or transfer request only a single byte/word.